cpp-client/deephaven/examples/demos/feedtimes.cc

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+ */
+#include <chrono>
+#include <condition_variable>
+#include <cstdlib>
+#include <exception>
+#include <iostream>
+#include <iomanip>
+#include <mutex>
+#include <optional>
+#include <sstream>
+#include <string>
+#include <thread>
+
+#include "deephaven/client/client.h"
+#include "deephaven/dhcore/types.h"
+#include "deephaven/dhcore/chunk/chunk.h"
+#include "deephaven/dhcore/column/column_source.h"
+#include "deephaven/dhcore/utility/utility.h"
+#include "deephaven/dhcore/utility/time.h"
+
+using deephaven::client::Client;
+using deephaven::client::TableHandle;
+using deephaven::client::TableHandleManager;
+using deephaven::dhcore::DateTime;
+using deephaven::dhcore::chunk::BooleanChunk;
+using deephaven::dhcore::chunk::DateTimeChunk;
+using deephaven::dhcore::column::DateTimeColumnSource;
+using deephaven::dhcore::container::RowSequence;
+using deephaven::dhcore::clienttable::ClientTable;
+using deephaven::dhcore::utility::VerboseCast;
+using deephaven::dhcore::utility::ToDateTime;
+using deephaven::dhcore::utility::ToTimePoint;
+using deephaven::dhcore::utility::FormatDuration;
+
+using my_clock = std::chrono::high_resolution_clock;
+using my_time_point = my_clock::time_point;
+using my_duration = my_clock::duration;
+
+namespace {
+void Run(const TableHandleManager &manager, const char *table_name, const char *col_name, int period_seconds);
+}  // namespace
+
+int main(int argc, char *argv[]) {
+  // avoid absl log initialization warnings
+  char grpc_env[] = "GRPC_VERBOSITY=ERROR";
+  putenv(grpc_env);
+  try {
+    if (argc != 5) {
+        std::cerr << "Usage: " << argv[0] << " <host:port> <table-name> <timestamp-column-name> <period-seconds>\n";
+        std::exit(1);
+      }
+
+    int c = 0;
+    const char *server = argv[++c];
+    const char *table_name = argv[++c];
+    const char *col_name = argv[++c];
+    const int period_seconds = std::stoi(argv[++c]);
+    auto client = Client::Connect(server);
+    auto manager = client.GetManager();
+
+    Run(manager, table_name, col_name, period_seconds);
+  } catch (const std::exception &e) {
+    std::cerr << "Caught exception: " << e.what() << '\n';
+    std::exit(1);
+  }
+  return 0;
+}
+
+namespace {
+
+const char *musecs = "\u03bcs";
+
+std::string FormatMicros(const uint64_t us) {
+  const uint64_t ms = us / 1'000UL;
+  const uint64_t rest_us = us % 1'000UL;
+  std::stringstream ss;
+  ss << ms;
+  if (rest_us > 0) {
+    ss << '.' << std::setw(3) << std::setfill('0') << rest_us;
+  }
+  ss << "ms (";
+  ss << us << musecs << ")";
+  return ss.str();
+}
+
+uint64_t usecs(const my_duration &d) {
```

Comment:
I have a micro-debate in my head about whether this should be `Usecs` (coding convention) or `usecs` (reads more naturally)